### PR TITLE
:globe_with_meridians: :flags: i18n integration

### PR DESCRIPTION
**Pull request by Cs4K1Sr4C**
- :globe_with_meridians: i18n module
- :mag: browser detection
- :hungary: Hungarian translation
- :rocket:  translator module - npm run translate

### CODE REVIEW

**Comment by miurla** [comment]
Amazing! Great stuff!  🙌

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This is a port of [babyagi](https://github.com/yoheinakajima/babyagi) with [Lang
 - [x] Execution history
 - [x] Faster speeds and fewer errors. ([😺 BabyCatAGI](https://twitter.com/yoheinakajima/status/1657448504112091136))
 - [ ] Display the current task and task list
-- [ ] i18n support
+- [x] i18n support
```
Comment (miurla) [inline README.md:24]: 🙌 🙌

File: src/components/LanguageSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
+import { SelectItem } from '@/types';
```
Comment (miurla) [inline src/components/LanguageSelector.tsx:1]: FYI: The language selection menu was implemented separately, so this file was removed in #40.  🙏  

https://github.com/miurla/babyagi-ui/pull/40

**Review by miurla — COMMENTED** [review]
Can the following i18next warnings be resolved?

```
i18next::translator: key "INFINITY" for languages "en" won't get resolved as namespace "common" was not yet loaded This means something IS WRONG in your setup. You access the t function before i18next.init / i18next.loadNamespace / i18next.changeLanguage was done. Wait for the callback or Promise to resolve before accessing it!!!
i18next::translator: missingKey en common INFINITY constants
i18next::translator: missingKey en common UNTIL_ALL_TASKS_COMPLETED constants
```

```
i18next: init: i18next is already initialized. You should call init just once!
```

File: package.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "next lint",
+    "translate": "node ./translate.cjs"
```
Comment (miurla) [inline package.json:10]: File name is wrong.

`translate.cjs` -> `translator.cjs`

File: translator.cjs [hunk 4]
```diff
@@ -0,0 +1,359 @@
+require('dotenv').config();
+const fs = require('fs');
+const path = require('path');
+const { Configuration, OpenAIApi } = require('openai');
+
+class Translator {
```
Comment (miurla) [inline translator.cjs:6]: 🆒

File: README.md (hunk 1 above)
Comment (Cs4K1Sr4C) [reply README.md:24]: There are no missing keys at the moment and the duplicated loading I think I've fixed. No Hydration errors.

**Comment by miurla** [comment]
Can you fix this build error?

Screenshot 2023-05-18 at 1 50 57

**Review by Cs4K1Sr4C — COMMENTED** [review]
PASS on local: `npm run build`

**Comment by miurla** [comment]
Thanks for the quick fix!

**Comment by miurla** [comment]
I was getting the following warning at build time, which probably caused the text to not output properly.
I have now done an instant rollback.


```
./node_modules/next-i18next/dist/esm/config/createConfig.js 02:30:33.673 A Node.js API is used (process.cwd at line: 141) which is not supported in the Edge Runtime. 02:30:33.673 Learn more: https://nextjs.org/docs/api-reference/edge-runtime
JAPANESE
./node_modules/next-i18next/dist/esm/config/createConfig.js
```

You can check the preview below.

- https://babyagi-45390ea68-babyagi-ui.vercel.app/

**Comment by Cs4K1Sr4C** [comment]
@miurla What should I do with this warning? Can you handle it or should I figure it out how can we resolve this? I've read the docs, but I couldn't find any exact fix for the appeared warning message. Also I've opened the preview deployment from a device which has 🇭🇺 language and I see only untranslated parts, but they are because those aren't translated yet. Maybe we can just avoid this warning...?...?

**Pull request merged**